### ISSUE TITLE
Phusion Passenger seems to stop before starting

### ISSUE THREAD

**Issue by RiteshAdobe** [phusion/passenger#1892]
Hi,

I have a ruby-based dockerized (docker version 1.12.1) web app (Ruby 2.3.0, RVM, Rails 4.2) and i am experiencing some hiccups while starting phusion passenger. It seems like Passenger tends to **STOP** before **STARTING**.

Here are the logs -

[ 2016-11-18 11:31:30.6807 5461/7f60405ed880 age/Wat/WatchdogMain.cpp:1291 ]: **Starting Passenger watchdog...**
[ 2016-11-18 11:31:30.6906 5464/7f3a925f2880 age/Cor/CoreMain.cpp:982 ]: **Starting Passenger core...**
[ 2016-11-18 11:31:30.6907 5464/7f3a925f2880 age/Cor/CoreMain.cpp:235 ]: **Passenger core running in multi-application mode.**
[ 2016-11-18 11:31:30.6929 5464/7f3a925f2880 age/Cor/CoreMain.cpp:732 ]: **Passenger core online, PID 5464**
[ 2016-11-18 11:31:30.7008 5471/7f04054bc880 age/Ust/UstRouterMain.cpp:529 ]: **Starting Passenger UstRouter...**
[ 2016-11-18 11:31:30.7032 5471/7f04054bc880 age/Ust/UstRouterMain.cpp:342 ]: **Passenger UstRouter online, PID 5471**

[ 2016-11-18 11:31:30.7260 5464/7f3a8c7a0700 age/Cor/CoreMain.cpp:532 ]: **Signal received. Gracefully shutting down... (send signal 2 more time(s) to force shutdown)**

[ 2016-11-18 11:31:30.7261 5471/7f03ffa4f700 age/Ust/UstRouterMain.cpp:422 ]: **Signal received. Gracefully shutting down... (send signal 2 more time(s) to force shutdown)**
[ 2016-11-18 11:31:30.7261 5471/7f04054bc880 age/Ust/UstRouterMain.cpp:492 ]: **Received command to shutdown gracefully. Waiting until all clients have disconnected...**
[ 2016-11-18 11:31:30.7262 5471/7f03ff24e700 Ser/Server.h:817 ]: [UstRouterApiServer] Freed 0 spare client objects
[ 2016-11-18 11:31:30.7262 5471/7f03ff24e700 Ser/Server.h:464 ]: [UstRouterApiServer] Shutdown finished
[ 2016-11-18 11:31:30.7262 5464/7f3a925f2880 age/Cor/CoreMain.cpp:901 ]: Received command to shutdown gracefully. Waiting until all clients have disconnected...
[ 2016-11-18 11:31:30.7262 5464/7f3a8bf9f700 Ser/Server.h:817 ]: [ApiServer] Freed 0 spare client objects
[ 2016-11-18 11:31:30.7263 5464/7f3a8bf9f700 Ser/Server.h:464 ]: [ApiServer] Shutdown finished
[ 2016-11-18 11:31:30.7263 5471/7f03ffa4f700 Ser/Server.h:464 ]: [UstRouter] Shutdown finished
[ 2016-11-18 11:31:30.7266 5464/7f3a8c7a0700 Ser/Server.h:817 ]: [ServerThr.1] Freed 128 spare client objects
[ 2016-11-18 11:31:30.7266 5464/7f3a8c7a0700 Ser/Server.h:464 ]: [ServerThr.1] Shutdown finished
[ 2016-11-18 11:31:30.7271 5471/7f04054bc880 age/Ust/UstRouterMain.cpp:523 ]: Passenger UstRouter shutdown finished
[Fri Nov 18 11:31:30.737076 2016] [ssl:warn] [pid 5485:tid 140365132089152] AH01873: Init: Session Cache is not configured [hint: SSLSessionCache]
[ 2016-11-18 11:31:30.7438 5486/7fd585acb880 age/Wat/WatchdogMain.cpp:1291 ]: **Starting Passenger watchdog...**
[ 2016-11-18 11:31:30.7532 5489/7f88c4268880 age/Cor/CoreMain.cpp:982 ]: **Starting Passenger core...**
[ 2016-11-18 11:31:30.7534 5489/7f88c4268880 age/Cor/CoreMain.cpp:235 ]: Passenger core running in multi-application mode.
[ 2016-11-18 11:31:30.7550 5489/7f88c4268880 age/Cor/CoreMain.cpp:732 ]: Passenger core online, PID 5489
[ 2016-11-18 11:31:30.7609 5496/7ff322e79880 age/Ust/UstRouterMain.cpp:529 ]: **Starting Passenger UstRouter...**
[ 2016-11-18 11:31:30.7618 5496/7ff322e79880 age/Ust/UstRouterMain.cpp:342 ]: Passenger UstRouter online, PID 5496
[ 2016-11-18 11:31:30.7620 5464/7f3a925f2880 age/Cor/CoreMain.cpp:967 ]: **Passenger core shutdown finished**
[Fri Nov 18 11:31:30.763428 2016] [core:warn] [pid 5485:tid 140365132089152] AH00098: pid file /apps/apache/logs/httpd.pid overwritten -- **Unclean shutdown of previous Apache run?**
[Fri Nov 18 11:31:30.764723 2016] [mpm_event:notice] [pid 5485:tid 140365132089152] AH00489: **Apache/2.4.23 (Unix) OpenSSL/1.0.1e-fips Phusion_Passenger/5.0.30 configured -- resuming normal operations**
[Fri Nov 18 11:31:30.764756 2016] [core:notice] [pid 5485:tid 140365132089152] AH00094: Command line: '/apps/apache/bin/httpd'

**I am using Passenger standalone 5.0.30/apache on CentOS - 7 .**

Steps to reproduce -

1) Install apache and passenger
2) start Apache with proper passenger config defined

**PS :- 
We have installed passenger using standard method  and stored the binaries in our local repository for re-use. I hope that shouldn't be a problem**

**Comment by OnixGH** [comment]
This is a known behavior caused by the way Apache loads plugins, and should not be an issue.

**Comment by RiteshAdobe** [comment]
Thanks a lot!

**Comment by alobban** [comment]
I am having this issue running Passenger Standalone (Sinatra) and Jenkins. When I incorporate the Passenger Stop/Start commands in my build. It says it's started however when I try to connect, I get a 502 bad request in my browser, but I get in my Passenger log:
2017/02/09 13:39:32 [crit] 12927#0: __*15 connect() to unix:/tmp/passenger.jvkpuSn/agents.s/core failed       (2: No such file or directory) while connecting to upstream, client: 192.168.1.85,__ server: _, request: "GET /ping HTTP/1.1", upstream: "passenger:unix:/tmp/passenger.jvkpuSn/agents.s/core:", host: "localhost".

Has any seen this issue and have a solution?

**Comment by CamJN** [comment]
If the core socket is missing that is a separate issue, please file a new issue with all the requested information.

**Comment by CamJN** [comment]
@sbourgeois1 please start a new ticket and provide all the information it asks for.
